app/src/main/java/org/glucosio/android/presenter/ExternalViewPresenter.java

```
@@ -36,13 +36,24 @@ public void onViewCreated() {
     if (network.isConnected()) {
       String title = view.extractTitle();
       String url = view.extractUrl();
+      parametersPrecondition(title, url);
       view.setupToolbarTitle(title);
       view.loadExternalUrl(url);
     } else {
       view.showNoConnectionWarning();
     }
   }
 
+  private void parametersPrecondition(String title, String url) {
+    if ((invalidParam(title)) || invalidParam(url)) {
+      throw new RuntimeException("Invalid arguments: need URL and TITLE");
```

I would throw `InvalidArgumetException`